components/Activity.tsx

```diff
@@ -1,19 +1,46 @@
-import { StyleSheet, TouchableOpacity, View } from "react-native"
+import { StyleSheet, TouchableOpacity, View, Modal } from "react-native"
+import { useState } from "react"
 import CustomText from "@/components/ui/CustomText"
 import EditIcon from "@/components/vectors/EditIcon"
+import EditActivity from "@/screens/EditActivity"
 
 export default function Activity() {
+  const [isModalVisible, setIsModalVisible] = useState(false)
+  const [activity, setActivity] = useState("Available")
+
+  const openModal = () => setIsModalVisible(true)
+  const closeModal = () => setIsModalVisible(false)
+
+  const updateAvailabilityText = (newText: string) => {
+    setActivity(newText)
+    closeModal()
+  }
+
   return (
     <View style={styles.container}>
-      <CustomText size="sm" fontWeight="medium" style={styles.signalText}>
-        Signal
+      <CustomText size="sm" fontWeight="medium" style={styles.titleText}>
+        Status
       </CustomText>
-      <View style={styles.availableContainer}>
-        <CustomText size="xl">Available</CustomText>
-        <TouchableOpacity onPress={() => ""}>
+      <View style={styles.statusContainer}>
+        <CustomText size="xl">{activity}</CustomText>
+        <TouchableOpacity onPress={openModal}>
           <EditIcon />
         </TouchableOpacity>
       </View>
+
+      {isModalVisible && (
+        <Modal
+          visible={isModalVisible}
+          transparent={true}
+          animationType="slide"
+          presentationStyle="pageSheet"
```

Comment:
opening the editScreen, we are getting a warning saying `Modal with 'pageSheet' presentation style and 'transparent' value is not supported`.  let's use ` presentationStyle="overFullScreen"` instead

```suggestion
          presentationStyle="overFullScreen"   
```